tests/fips/openssl/openssl_fips_alglist.pm

```diff
@@ -15,64 +15,75 @@ use base "consoletest";
 use testapi;
 use strict;
 use warnings;
-use utils 'zypper_call';
-use version_utils qw(is_sle is_transactional);
+use serial_terminal 'select_serial_terminal';
+use version_utils qw(is_sle is_sle_micro is_transactional);
+use security::openssl_misc_utils;
 
-sub run {
-    select_console 'root-console';
-
-    # openssl pre-installed in SLE Micro
-    zypper_call 'in openssl' unless is_transactional;
-    zypper_call('info openssl');
-    my $current_ver = script_output("rpm -q --qf '%{version}\n' openssl");
-
-    # openssl attempt to update to 1.1.1+ in SLE15 SP4 base on the feature
-    # SLE-19640: Update openssl 1.1.1 to current stable release
-    if (!is_sle('<15-sp4') && ($current_ver ge 1.1.1)) {
-        record_info('openssl version', "Version of Current openssl package: $current_ver");
-    }
-    else {
-        record_soft_failure('jsc#SLE-19640: openssl version is outdated and need to be updated over 1.1.1+ for SLE15-SP4');
-    }
-
-    # Seperate the diffrent openssl command usage between SLE12 and SLE 15-SP{0..5} and SLE 15-SP6+
-    if (is_sle('<15')) {
-        # List message digest algorithms in fips mode
-        # only SHA1 and SHA2 (224, 256, 384, 512) are approved in fips mode
-        # Note: DSA is short of DSA-SHA1, so it is also valid item
+sub test_openssl1 {
+    my $openssl_binary = shift // "openssl";
+    eval {
```

Comment:
why do you need `eval`  ?

Reply:
this has always been there, but we can definitely remove it!